fabtests/functional/unexpected_msg.c

```diff
@@ -199,6 +200,96 @@ static int run_test_loop(void)
 	return ret;
 }
 
+static int exchange_unexp_addr(void)
+{
+	char temp[FT_MAX_CTRL_MSG];
+	size_t addrlen;
+	int ret;
+
+	if (opts.dst_addr) {
+		ret = ft_av_insert(av, fi->dest_addr, 1, &remote_fi_addr, 0,
+				   NULL);
+		if (ret)
+			goto err;
+
+		addrlen = FT_MAX_CTRL_MSG;
+		ret = fi_getname(&ep->fid, temp, &addrlen);
+		if (ret)
+			goto err;
+
+		ret = ft_hmem_copy_to(opts.iface, opts.device,
+				      tx_buf + ft_tx_prefix_size(), temp,
+				      addrlen);
+		if (ret)
+			goto err;
+
+		/*
+		 * Send name twice - first will be matched to FI_ADDR_UNSPEC
+		 * Second will be matched to directed receive after fi_av_insert
+		 */
+		ret = ft_post_tx_buf(ep, remote_fi_addr, addrlen, 0, &tx_ctx,
+				     tx_buf, mr_desc, ft_tag);
+		if (ret)
+			goto err;
+
+		ret = ft_post_tx_buf(ep, remote_fi_addr, addrlen, 0, &tx_ctx,
+				     tx_buf, mr_desc, ft_tag);
+		if (ret)
+			return ret;
+
+		ft_sync();
+
+		/* Make sure server can send back to us */
+		ret = ft_post_rx(ep, rx_size, &rx_ctx);
+		if (ret)
+			goto err;
+
+		ret = ft_get_rx_comp(rx_seq);
+		if (ret)
+			goto err;
+	} else {
+		ft_sync();
+
+		/* Process first unexpected message with unspec addr*/
+		ret = ft_post_rx(ep, rx_size, &rx_ctx);
```

Comment:
there was a rx posted in ft_ep_enable_recv. Do we need this ?

Reply:
the pre posted receive only happens if we use the common code message allocation which is avoided in this test by setting
FT_OPT_SKIP_MSG_ALLOC

Reply:
I see, I missed that we have a separate buf in this test.